libs/server/Resp/BasicCommands.cs

```diff
@@ -610,69 +562,56 @@ private bool NetworkSETEXNX<TGarnetApi>(ref TGarnetApi storageApi)
             return true;
         }
 
-        private bool NetworkSET_EX<TGarnetApi>(RespCommand cmd, int expiry, byte* keyPtr, byte* valPtr,
-            int vsize, bool highPrecision, ref TGarnetApi storageApi)
+        private unsafe bool NetworkSET_EX<TGarnetApi>(RespCommand cmd, int expiry, ref SpanByte key, ref SpanByte val,
+            bool highPrecision, ref TGarnetApi storageApi)
             where TGarnetApi : IGarnetApi
         {
             Debug.Assert(cmd == RespCommand.SET);
 
-            if (expiry == 0) // no expiration provided - this code path will not be currently hit as TrySET is used
-            {
-                *(int*)valPtr = vsize;
-            }
-            else
+            if (expiry != 0)
             {
-                // Move payload forward to make space for metadata
-                Buffer.MemoryCopy(valPtr + sizeof(int), valPtr + sizeof(int) + sizeof(long), vsize, vsize);
-                *(int*)valPtr = vsize + sizeof(long);
-                SpanByte.Reinterpret(valPtr).ExtraMetadata = DateTimeOffset.UtcNow.Ticks +
-                                                             (highPrecision
-                                                                 ? TimeSpan.FromMilliseconds(expiry).Ticks
-                                                                 : TimeSpan.FromSeconds(expiry).Ticks);
+                var newValLen = val.Length + sizeof(long);
+                var newVal = stackalloc byte[newValLen];
```

Comment:
We cannot stackalloc such a large buffer. stackalloc should be used for a small and known-up-front constant bytes of stack data.

Reply:
Please check other uses of stackalloc as well.